### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by acaire**
Attempt to address #82

*Description of changes:*
Adds a Helm chart for aws-service-operator and correct a minor typo with some perceived brand consistency.  Happy to drop the last commit if necessary - I haven't tested it heavily, feel free to change anything you need to :)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by acaire** [comment]
Just noticed i'm missing `NOTES.txt` - Will work on this tomorrow, until then feedback appreciated, thanks!

**Comment by KierranM** [comment]
👍 Had just started making this helm chart myself in the helm/charts repo before I seen this. Quick question, how/where will consumers(me in this case) of the chart access it from? One of the main advantages of putting charts in the helm/charts repo is that you get that for free. Do y'all have your own helm repo?

File: charts/aws-service-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+## Configuration
+
+The following table lists the configurable parameters of the aws-service-operator chart and their default values.
+
+| Parameter                 | Description                            | Default                                            |
+| ------------------------- | -------------------------------------- | -------------------------------------------------- |
+| `image.repository`        | Container image repository             | `awsserviceoperator/aws-service-operator`          |
+| `image.tag`               | Container image tag                    | `v0.0.1-alpha2`                                    |
+| `image.pullPolicy`        | Container pull policy                  | `IfNotPresent`                                     |
+| `operator.accountId       | AWS Account ID to operator on          | `""`                                               |
```
Comment (christopherhein) [inline charts/aws-service-operator/README.md]: Lines 27-35 prefixed with `operator` need a closing tick mark after the key. Check the rendered output to see.

File: charts/aws-service-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "aws-service-operator.fullname" . }}
+  labels:
+    app: {{ template "aws-service-operator.name" . }}
+    chart: {{ template "aws-service-operator.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
```
Comment (christopherhein) [inline charts/aws-service-operator/templates/deployment.yaml]: These are reused in a few places, can we make a helper template for them?

**Comment by christopherhein** [comment]
> 👍 Had just started making this helm chart myself in the helm/charts repo before I seen this. Quick question, how/where will consumers(me in this case) of the chart access it from? One of the main advantages of putting charts in the helm/charts repo is that you get that for free. Do y'all have your own helm repo?

I think we should copy it over once we have this stable, long-term there should be a release mechanism built into a CI process having the source in the repo will help as features like #107 get finalized which drastically change the way the CRD manifests are handled.

File: charts/aws-service-operator/templates/deployment.yaml (hunk 2 above)
Comment (acaire) [reply charts/aws-service-operator/templates/deployment.yaml]: Yep, i'll also factor in the new naming standard: https://github.com/helm/helm/issues/4335

**Comment by christopherhein** [comment]
They should be now @davidxjohnson and @acaire, the latest release removed `operator-kit` which was managing the CRDs now we have them `config/aws-service-operator.yaml`

**Comment by davidxjohnson** [comment]
I was anticipating the files in `examples/cloudformationtemplates` to be included in the chart since they are needed by the aws-operator.

**Comment by christopherhein** [comment]
@davidxjohnson  by default they actually aren’t required, there is a fallback mechanism to an S3 bucket which holds the canonical version of the assets and will try to load from that bucket after checking the user provided bucket.

**Comment by davidxjohnson** [comment]
I see that the CRD manifests are in `configs/aws-service-operator.yaml`  ... but wondering if they should be in `charts/aws-service-operator/` ??? ... didn't see them in the list of file changes.

**Comment by christopherhein** [comment]
Yes @davidxjohnson those will need to be moved into the charts as well. They weren't included because when this was first authored the CRDs were deployed by the go application which they aren't now. See https://github.com/awslabs/aws-service-operator/pull/107 for more details

**Comment by davidxjohnson** [comment]
I'll wait for the helm chart work to merge and will look at it again then.

Regards

**Comment by acaire** [comment]
@christopherhein Added these manually, is it worth trying to do it programatically i.e. create a separate `templ` file to be generated inside the charts/template directory?

**Comment by acaire** [comment]
@christopherhein I've made these changes - Can you please let me know if anything further is required?

**Comment by christopherhein** [comment]
Thanks @acaire really appreciate this work! We're all merged, I'll be cutting a new build in the coming week and we'll have a tar.gz that you should be able to use to `helm install URL` from. For now anyone that finds this PR you can clone the repo and `helm install charts/aws-service-operator`

**Pull request merged**